test/Servant/JSSpec.hs

```diff
@@ -152,6 +157,21 @@ angularSpec test = describe specLabel $ do
         ngOpts = NG.defAngularOptions { NG.serviceName = testName }
         genJS req = NG.angularService ngOpts req
 
+instance Arbitrary T.Text where
```

Comment:
I'd have a newtype (`ASCII`) over `Text`, since for other tests we don't want to restrict the arbitrary to just ASCII. (And then just change below to `toValidFunctionName (getASCII x)`).